lib/util/host-rules.ts

```diff
@@ -153,6 +153,21 @@ export function find(search: HostRuleSearch) {
   return res;
 }
 
+export function getPlatformByHostOrUrl(url) {
```

Comment:
Is this used?

Reply:
Yes it does, however I forgot to pull that logic from another fork, here is the current state of affairs:
https://github.com/ocadotechnology/renovate/blob/04fb68896d803688ac6c094edbeebebc05615826/lib/datasource/index.ts#L142
You did mention that you wish for the flow to be something along the lines of:
1) Check if this is a valid public repo, done via 'hosted-git-info'
2) Check if there is a host rule for the relevant hostname.
Fixed.